### PR TITLE
bpo-39527: Update doc of argparse

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue39527

### CODE REVIEW

File: Doc/library/argparse.rst [hunk 1]
```diff
@@ -11,6 +11,10 @@
 
 **Source code:** :source:`Lib/argparse.py`
 
+.. testsetup::
+
+   import argparse
+
```
Comment (JulienPalard) [inline Doc/library/argparse.rst:17]: Adding a testsetup looks useless if you don't add any test.

File: Doc/library/argparse.rst [hunk 2]
```diff
@@ -660,7 +662,7 @@ Normally, when you pass an invalid argument list to the :meth:`~ArgumentParser.p
 method of an :class:`ArgumentParser`, it will exit with error info.
 
 If the user would like catch errors manually, the feature can be enable by setting
-``exit_on_error`` to ``False``::
+``exit_on_error`` to ``False``:
```
Comment (JulienPalard) [inline Doc/library/argparse.rst:665]: Why removing this block syntax?

**Review by JulienPalard — COMMENTED** [review]
I'm not sure this enhance the documentation. It's shorter, but on the other hand we'll have reports of "I copied it and it does not work" (when they forget the import).

**Comment by rhettinger** [comment]
Thanks for the suggestion, but I'm going to reject this one.

It only saves a couple of lines, but comes at the cost of the examples not being stand-alone, suitable for cut-and-paste into working code.

File: Doc/library/argparse.rst (hunk 2 above)
Comment (shihai1991) [reply Doc/library/argparse.rst:665]: `::` would stop running this test case(i don't know the reason, just from the actual doctest running result), so I delete a colon.

**Comment by shihai1991** [comment]
> Thanks for the suggestion, but I'm going to reject this one.
> 
> It only saves a couple of lines, but comes at the cost of the examples not being stand-alone, suitable for cut-and-paste into working code.

Thanks for your comment, raymond.

**Pull request closed without merging**